x/init.go

```diff
@@ -42,45 +39,13 @@ var (
 	lastCommitTime string
 )
 
-// SetTestRun sets a variable to indicate that the current execution is a test.
-func SetTestRun() {
-	isTest = true
-}
-
 // check if any version is set by ldflags. If not so, it should be set as "dev"
 func checkDev() {
 	if dgraphVersion == "" {
 		dgraphVersion = "dev"
 	}
 }
 
-// IsTestRun indicates whether a test is being executed. Useful to handle special
-// conditions during tests that differ from normal execution.
-func IsTestRun() bool {
-	return isTest
-}
-
-// AddInit adds a function to be run in x.Init, which should be called at the
-// beginning of all mains.
-func AddInit(f func()) {
-	initFunc = append(initFunc, f)
-}
-
-// Init initializes flags and run all functions in initFunc.
-func Init() {
```

Comment:
This definitely shouldn't be removed.
If you remove this it will break the Alphas this is called at /root/dev/dgraph/dgraph/cmd/alpha/run.go


Also there is a bunch of tests the relies on this.
ALSO, Ratel depends on `checkDev()` if you remove this it will break any Ratel that is connecting to any cluster that doesn't has a version release. Which means all unreleased binaries, tests and so on.

If we need to change this behavior it should have a plan. As Joshua pointed out.

IMO, all PRs should be single purpose. Small things could be ignored(like identation, or grammar). But a PR shouldn't have more than one Goal. Unless all changes are releated. 

For example, accepting this PR just to remove the WIKI path would be accepted easily, no-brainer. It was supposed to be merged. But these second changes should be made in a new PR. To also be discussed and identified the side effects of removals.

Reply:
Good catch Michel, I have fixed it now.